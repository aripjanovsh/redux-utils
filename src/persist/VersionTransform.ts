import { invariant } from "../internal/utils";
import { TransformConfig } from "./PersistConfigBuilder";

interface Plain<S> {
  r: S;
  v: string;
}

export function createVersionTransformConfig<S>(
  version: string,
): TransformConfig<S, Plain<S>> {
  invariant(
    version != null,
    "createVersionTransformConfig: `version` expected to be defined.",
  );

  return {
    in(state) {
      return { v: version, r: state };
    },

    out(plain) {
      return !plain || plain.v !== version ? null : plain.r;
    },
  };
}
