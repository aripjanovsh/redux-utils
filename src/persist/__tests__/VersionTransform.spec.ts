import { createVersionTransformConfig } from "../VersionTransform";

describe("VersionTransform", () => {
  test("init validation", () => {
    expect(() => {
      // @ts-ignore
      createVersionTransformConfig(null);
    }).toThrowErrorMatchingSnapshot();

    expect(() => {
      // @ts-ignore
      createVersionTransformConfig(undefined);
    }).toThrowErrorMatchingSnapshot();
  });

  test("transformation", () => {
    const config = createVersionTransformConfig("foo");

    expect(config.in("bar")).toEqual({ v: "foo", r: "bar" });

    expect(config.out({ v: "foo", r: "bar" })).toBe("bar");
    expect(config.out({ v: "baz", r: "bar" })).toBe(null);
  });
});
