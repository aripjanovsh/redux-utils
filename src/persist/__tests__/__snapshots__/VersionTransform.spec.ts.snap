// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionTransform init validation 1`] = `"createVersionTransformConfig: \`version\` expected to be defined."`;

exports[`VersionTransform init validation 2`] = `"createVersionTransformConfig: \`version\` expected to be defined."`;
